packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4333,66 +4335,24 @@ export class ITxn extends Op {
 		this.idx = undefined;
 
 		this.assertITxFieldDefined(args[0], interpreter.tealVersion, line);
-		if (TxArrFields[interpreter.tealVersion].has(args[0])) {
+		if (
+			TxArrFields[interpreter.tealVersion].has(args[0]) ||
```

Comment:
In case we are processing the inner transaction. Why should we find if there exist transactions here?

Reply:
Because we are not sure if there was already an inner transaction. This opcode can be invoke whenever we want